### PR TITLE
Set owner references for configmaps

### PR DESCRIPTION
**Pull request by rabi**
This ensures that they're garbage collected.

### CODE REVIEW

File: controllers/keystoneapi_controller.go [hunk 1]
```diff
@@ -723,7 +723,10 @@ func (r *KeystoneAPIReconciler) generateServiceConfigMaps(
 
 // reconcileConfigMap -  creates clouds.yaml
 // TODO: most likely should be part of the higher openstack operator
-func (r *KeystoneAPIReconciler) reconcileConfigMap(ctx context.Context, instance *keystonev1.KeystoneAPI) error {
+func (r *KeystoneAPIReconciler) reconcileConfigMap(
```
Comment (abays) [inline controllers/keystoneapi_controller.go:726]: Should this function instead use `lib-common`'s config map functions like the `KeystoneAPI` controller already is here: https://github.com/openstack-k8s-operators/keystone-operator/blob/d3b76057f74302810cf3e299421f9785a3ac3491/controllers/keystoneapi_controller.go#L674-L722?

Comment (rabi) [reply controllers/keystoneapi_controller.go:726]: Probably we should. maybe in a follow-up(?). I keep bug fixes and refactoring separate so as not to introduce more bugs.

Comment (rabi) [reply controllers/keystoneapi_controller.go]: I've updated it to use lib-common instead. The CI failure does not seem related.

Comment (rabi) [reply controllers/keystoneapi_controller.go]: CI issue resolved.

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**